src/jit/flowgraph.cpp

```diff
@@ -9315,7 +9315,7 @@ VARSET_VALRET_TP Compiler::fgGetVarBits(GenTreePtr tree)
     {
         VarSetOps::AddElemD(this, varBits, varDsc->lvVarIndex);
     }
-    else if (varDsc->lvType == TYP_STRUCT && varDsc->lvPromoted)
+    else if (tree->gtType == TYP_STRUCT && varDsc->lvPromoted)
```

Comment:
What is the case where this was doing the wrong thing? Was it a `GT_LCL_FLD` tree that should not be affecting all the fields?